app/(tabs)/groups.tsx

```diff
@@ -1,79 +1,34 @@
 import { router } from 'expo-router';
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useContext } from 'react';
 import { useTranslation } from 'react-i18next';
-import { ScrollView, Text, View } from 'react-native';
+import { TouchableOpacity, View } from 'react-native';
 
-import { CustomButton, TouchableExpense } from '@/components';
 import SafeView from '@/components/ui/box/SafeView';
-import useExpenses from '@/hooks/expense/UseExpenses';
-import useGroups, { Group } from '@/hooks/group/UseGroups';
+import Loader from '@/components/ui/loader/Loader';
+import MultiTextInput from '@/components/ui/text-input/MultiTextInput';
+import { GlobalContext } from '@/context/GlobalContext';
+import useGroup from '@/hooks/group/UseGroup';
 
 export default function Groups() {
   const { t } = useTranslation();
-  const { data, isFetching } = useGroups();
-  const [currentGroupId, setCurrentGroupId] = useState<string | null>(null);
-  const [groups, setGroups] = useState<Group[]>([]);
-  const { data: expensesData, isFetching: isFetchingExpenses } = useExpenses(currentGroupId);
-
-  function getNameById(id: string, items: Group[]): string | undefined {
-    const item = items.find((item) => item.groupId === id);
-    return item ? item.name : undefined;
-  }
-
-  useEffect(() => {
-    if (data && data.groups.length > 0) {
-      setGroups(data.groups);
-      setCurrentGroupId(data.groups[0].groupId);
-    }
-  }, [data]);
+  const { userData } = useContext(GlobalContext);
+  const { data: groupDetails } = useGroup(userData.currentGroupId);
 
   return (
     <SafeView>
-      {isFetching ? (
-        <Text>Loading...</Text>
-      ) : groups.length === 0 ? (
-        <>
-          <Text className="text-title2">{t("You don't belong to any group!")}</Text>
-          <View className="py-[200px] w-full flex flex-col justify-center items-center space-y-[32px]">
-            <View className="w-full">
-              <CustomButton
-                onPress={() => router.push('/(stepper)/(group-creation)/group-name')}
-                title={t('Create new group')}
-              />
-            </View>
-            <View className="w-full">
-              <CustomButton
-                onPress={() => router.push('/(popover)/(join-group)/join-group')}
-                title={t('Join existing group')}
-              />
-            </View>
-          </View>
-        </>
-      ) : (
-        <>
-          <Text className="text-title1">{getNameById(currentGroupId!, groups)}</Text>
-          {isFetchingExpenses ? (
-            <Text>Loading Expenses...</Text>
-          ) : expensesData?.expenses.length === 0 ? (
-            <Text>{t('No expenses')}</Text>
+      <View className="py-8">
+        {userData.currentGroupId ? (
```

Comment:
we can user !!userData.currentGroupId && , without using null later ( or sth like that)